templates/shared_services/sonatype-nexus/template_schema.json

```
@@ -5,5 +5,114 @@
   "title": "Nexus Shared Service",
   "description": "Provides Nexus shared service",
   "required": [],
-  "properties": {}
+  "properties": {},
+  "pipeline": {
+    "install": [
+      {
+        "stepId": "main"
+      },
+      {
+        "stepId": "a60421b3-7308-491f-b531-e007cdc0ff46",
+        "stepTitle": "Add nexus rule collection to firewall",
+        "resourceTemplateName": "tre-shared-service-firewall",
+        "resourceType": "shared-service",
+        "resourceAction": "upgrade",
+        "properties": [
+          {
+            "name": "rule_collections",
+            "type": "array",
+            "arraySubstitutionAction": "replace",
+            "arrayMatchField": "name",
+            "value": {
+              "name": "arc-web_app_subnet_nexus",
+              "action": "Allow",
+              "rules": [
+                {
+                  "name": "nexus-package-sources",
+                  "description": "Nexus Package Sources",
+                  "protocols": [
+                    {
+                      "port": "443",
+                      "type": "Https"
+                    },
+                    {
+                      "port": "80",
+                      "type": "Http"
+                    }
+                  ],
+                  "target_fqdns": "{{ resource.properties.nexus_allowed_fqdns_list }}",
+                  "source_addresses": "{{ resource.properties.address_prefixes }}"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    ],
+    "upgrade": [
+      {
+        "stepId": "main"
+      },
+      {
+        "stepId": "a60421b3-7308-491f-b531-e007cdc0ff47",
+        "stepTitle": "Update nexus rule collection in firewall",
+        "resourceTemplateName": "tre-shared-service-firewall",
+        "resourceType": "shared-service",
+        "resourceAction": "upgrade",
+        "properties": [
+          {
+            "name": "rule_collections",
+            "type": "array",
+            "arraySubstitutionAction": "replace",
+            "arrayMatchField": "name",
+            "value": {
+              "name": "arc-web_app_subnet_nexus",
```

nit: is there a particular reason why mixed -_? is arc = azure rule collection (just interested!) 
```suggestion
              "name": "arc-web_app_subnet_nexus",
```

---

no reason, that's just what it was called in the TF, so used the same name to avoid disruption